src/relay/backend/contrib/tensorrt/enable_tensorrt.cc

```diff
@@ -0,0 +1,569 @@
+      : trt_version_(trt_version) {}
+
+  void VisitExpr_(const VarNode* op) {
+    const auto* ttype = op->checked_type().as<TensorTypeNode>();
+    CHECK(ttype);
+    if (!ttype->dtype.is_float()) {
+      compatible_ = false;
+    }
+  }
+
+  void VisitExpr_(const CallNode* call) final {
+    const std::string op_name = (call->op.as<OpNode>())->name;
+    for (size_t i = 0; i < call->args.size(); ++i) {
+      // Workaround for check for transpose of weight. This check allows weights
+      // to be either 1) VarNode 2) ConstantNode 3) VarNode or Constant Node
+      // with a transpose between. This transpose is allowed to have permuation
+      // which isn't allowed by TensorRT since we will process this transpose
+      // manually.
+      // This workaround isn't required anymore since ConstantFolding will take
+      // care of the transpose for us. However, in the case where the weights
+      // aren't marked as params it can still be useful.
+      if ((op_name == "nn.conv2d" || op_name == "nn.dense") && i == 1) {
+        auto* transpose = call->args[i].as<CallNode>();
+        if (transpose && transpose->op.as<OpNode>()->name == "transpose") {
+          if (!transpose->args[0].as<VarNode>() &&
+              !call->args[i].as<ConstantNode>()) {
+            compatible_ = false;
+            LOG(INFO) << op_name
+                      << " not supported: must have constant weight.";
+          }
+        } else if (!call->args[i].as<VarNode>() &&
+                   !call->args[i].as<ConstantNode>()) {
+          compatible_ = false;
+          LOG(INFO) << op_name << " not supported: must have constant weight.";
+        }
+      } else {
+        VisitExpr(call->args[i]);
+      }
+    }
+    auto it = trt_compatible_ops.find(op_name);
+    if (it == trt_compatible_ops.end() ||
+        !it->second(call, op_name, trt_version_)) {
+      LOG(INFO) << op_name << " not supported.";
+      compatible_ = false;
+    }
+  }
+
+  bool Check(const Expr& expr) {
+    compatible_ = true;
+    VisitExpr(expr);
+    return compatible_;
+  }
+
+ private:
+  bool compatible_;
+  std::tuple<int, int, int> trt_version_;
+};
+
+/*!
+ * \brief Modifies a relay expr so that TensorRT codegen will be used.
+ */
+class TrtEnabler : public ExprMutator {
+ public:
+  explicit TrtEnabler(const std::tuple<int, int, int>& trt_version)
+      : trt_version_(trt_version) {}
+
+  Expr VisitExpr_(const VarNode* op) {
+    // Rename inputs
+    auto var = VarNode::make("tensorrt_input" + std::to_string(var_id_++),
+                             op->checked_type_);
+
+    original_inputs_.push_back({var, GetRef<Expr>(op)});
+    return std::move(var);
+  }
+
+  Expr Enable(const Expr& expr) {
+    // Get inputs to func.
+    auto func = expr.as<FunctionNode>();
+    CHECK(func != nullptr);
+    Array<Var> func_params;
+    for (auto param : func->params) {
+      func_params.push_back(param);
+    }
```

Comment:
Can you use constructor(vector) or constructor(begin, end) to create and fill `func_params` here?